### PR TITLE
Backports v0.18- PR 13769 and 13767

### PR DESCRIPTION
**Pull request by pravblockc**
Backport the below mentioned PR's from o.18 version to Dash dev branch

Merge #13769: Mark single-argument constructors "explicit"
Merge #13767: Remove redundant assignments (dead stores)

### CODE REVIEW

File: src/interfaces/wallet.cpp [hunk 1]
```diff
@@ -177,8 +177,7 @@ class WalletImpl : public Wallet
 {
 public:
     CoinJoinImpl m_coinjoin;
-
-    WalletImpl(const std::shared_ptr<CWallet>& wallet) : m_shared_wallet(wallet), m_wallet(*wallet.get()), m_coinjoin(*wallet.get()) {}
+    explicit WalletImpl(const std::shared_ptr<CWallet>& wallet) : m_shared_wallet(wallet), m_wallet(*wallet.get()) {}
```
Comment (PastaPastaPasta) [inline src/interfaces/wallet.cpp]: Should retain this newline imo

File: src/test/validation_block_tests.cpp [hunk 2]
```diff
@@ -23,7 +23,7 @@ BOOST_FIXTURE_TEST_SUITE(validation_block_tests, RegtestingSetup)
 struct TestSubscriber : public CValidationInterface {
     uint256 m_expected_tip;
 
-    TestSubscriber(uint256 tip) : m_expected_tip(tip) {}
+    explicit TestSubscriber(uint256 tip) : m_expected_tip(tip) {}
```
Comment (PastaPastaPasta) [inline src/test/validation_block_tests.cpp:26]: 13769 https://github.com/bitcoin/bitcoin/pull/13769/files#diff-b12066a688b85bea440eb38f079fb8ecc1984318470631bca7e3547a98effaa9L32-R32 is missing. HasReason is in test_dash.h

File: src/test/script_tests.cpp [hunk 3]
```diff
@@ -895,17 +895,19 @@ BOOST_AUTO_TEST_CASE(script_build)
         }
     }
 
+#ifdef UPDATE_JSON_TESTS
```
Comment (PastaPastaPasta) [inline src/test/script_tests.cpp:898]: 13767 added empty file descriptor_tests.cpp (yet the changes in descriptor_tests are needed and missing https://github.com/dashpay/dash/blob/develop/src/test/descriptor_tests.cpp#L67)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/test/validation_block_tests.cpp (hunk 2 above)
Comment (pravblockc) [reply src/test/validation_block_tests.cpp:26]: Brought in missing changes

File: src/test/script_tests.cpp (hunk 3 above)
Comment (pravblockc) [reply src/test/script_tests.cpp:898]: Fixed

**Comment by PastaPastaPasta** [comment]
Looks good :+1: 

A couple of things

1. Avoid merging in develop, instead you should `git rebase`. Please do this and drop the merge commit. 
2. Please squash the changes relevant to each PR into that backport PR.

**Comment by pravblockc** [comment]
New PR has been raised for this: https://github.com/dashpay/dash/pull/4317 to avoid conflicts in git history

**Pull request closed without merging**